src/control/ros/RosTrajectoryExecutor.cpp

```diff
@@ -104,10 +107,20 @@ void RosTrajectoryExecutor::validate(const trajectory::Trajectory* traj)
       traj->getStateSpace());
 
   if (!space)
+  {
     throw std::invalid_argument(
         "Trajectory is not in a MetaSkeletonStateSpace.");
+  }
+
+  dart::dynamics::MetaSkeletonPtr metaSkeleton
+      = dart::dynamics::Group::create("check", getDofs());
+  if (!space->isCompatible(metaSkeleton.get()))
```

Comment:
@amalnanavati Thank you for checking on real! The exception I believe that gets triggered should be this one (rather than the one is Instantaneous Traj Executor). I'll double-check that.